### PR TITLE
Remove null checks

### PR DESCRIPTION
**Pull request by EmmaWenzel**
Remove null checks in game utils functions and some user utils functions and test for those functions throwing exceptions instead.

### CODE REVIEW

File: src/test/java/com/google/sps/GameUtilTest.java [hunk 1]
```diff
@@ -61,9 +61,12 @@ public void createGame_EmptyGameName() {
...
 
-    Entity actual = GameUtils.createGameEntity("game", null);
-
-    Assert.assertEquals(null, actual);
+    Entity actual = null;
+    try{
+        actual = GameUtils.createGameEntity("game", null);
+    }catch(NullPointerException e){
+        Assert.assertEquals(null, actual);
+    }
```
Comment (devonwasson) [inline src/test/java/com/google/sps/GameUtilTest.java:69]: here (and everywhere else in this PR) we'll want to actually assert that the exception was thrown.

What you did is technically a correct way of accomplishing this goal, but it's not intuitive and there is a more elegant way to do this:

https://howtodoinjava.com/junit5/expected-exception-example/

Let me know if you have any questions!

File: src/main/java/com/google/sps/servlets/startGameServlet.java [hunk 2]
```diff
@@ -45,6 +45,11 @@ public startGameServlet(){
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
  
     String gameName = request.getParameter("game-name");
+
+    if(gameName == ""){
```
Comment (devonwasson) [inline src/main/java/com/google/sps/servlets/startGameServlet.java:49]: Might want to change to gameName.isEmpty()

File: src/test/java/com/google/sps/GameUtilTest.java (hunk 1 above)
Comment (EmmaWenzel) [reply src/test/java/com/google/sps/GameUtilTest.java]: Okay, that's done.

File: src/main/java/com/google/sps/servlets/startGameServlet.java (hunk 2 above)
Comment (EmmaWenzel) [reply src/main/java/com/google/sps/servlets/startGameServlet.java]: Got it; changed.

**Pull request merged**